src/containers/DataSet/index.js

```diff
@@ -37,61 +57,131 @@ class DataSet extends React.Component {
-      return <DataSetReady {...props} />;
-    } else {
-      return <DataSetExpired />;
-    }
-  } else {
-    // 2. if it's not ready to be downloaded, then allow the user to set an email and receive an alert when its ready
-    if (!email_address) {
-      return <DatasetNoEmail {...props} />;
+class DataSetPage extends React.Component {
+  state = {
+    changedEmail: false
+  };
+
+  handleEmailChange = () => {
+    this.setState({
+      changedEmail: true
+    });
+  };
+
+  render() {
+    const {
+      is_processed,
+      is_available,
+      email_address,
+      s3_bucket,
+      s3_key,
+      dataSetId,
+      ...props
+    } = this.props;
+    // 1. Check if the dataset is already processed, if true show a link to the download file
+    if (is_processed) {
+      if (is_available) {
+        return <DataSetReady {...props} />;
+      } else {
+        return <DataSetExpired />;
+      }
     } else {
-      // 3. Allow the user to change its email if it's already added
-      return <DataSetWithEmail {...props} email={email_address} />;
+      // 2. if it's not ready to be downloaded, then allow the user to set an email and receive an alert when its ready
+      if (!email_address) {
+        if (!this.state.changedEmail) {
+          return <DatasetNoEmail {...props} />;
+        } else {
+          // 3. Allow the user to change its email if it's already added
+          return (
+            <DataSetWithEmail
+              {...props}
+              email={email_address}
+              handleSubmit={this.handleEmailChange}
+            />
+          );
+        }
+      } else {
+        return (
+          <DataSetProcessing email={email_address} dataSetId={dataSetId} />
+        );
+      }
     }
   }
 }
 
 /**
  * This component gets rendereded in the DataSet page, when no email has been assigned
  */
-function DatasetNoEmail({ id }) {
-  return (
-    <div>
-      <h1>
-        We’re putting your download file together. It usually takes 15- 20
-        minutes.
-      </h1>
-      <h2>
-        Enter your email and we will email you when the files are ready for
-        download.
-      </h2>
-
-      <EmailForm dataSetId={id} />
-
-      <div className="dataset__image">
-        <img src={ProcessingImage} alt="" />
+class DatasetNoEmail extends React.Component {
+  state = {
+    agreedToTerms: false,
+    token: null
+  };
+
+  componentDidMount() {
+    const token = localStorage.getItem('refinebio-token');
+    if (!!token) {
+      this.setState({ token });
+    }
+  }
+
+  handleAgreedToTerms = () => {
+    this.setState({ agreedToTerms: !this.state.agreedToTerms });
+  };
+
+  render() {
+    const { id, startDownload } = this.props;
+    return (
+      <div>
+        <h1>
+          We're ready to start putting your download files together. It usually
+          takes about 15-20 minutes.
+        </h1>
+        <h2>
+          Enter your email and we will email you the download link when your
```

Comment:
It may sound a little better to use `send you` instead of `email you`, because the first `email` is being used as a noun. (English is not my native language, so please feel free to leave it as is.)